src/components/SignUpForm.js

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import ErrorMsg from "./ErrorMsg";
+import { Formik, Field, ErrorMessage, Form } from "formik";
+import * as Yup from "yup";
+import styled from "styled-components";
+const initialValues = {
+  fullName: "",
+  email: "",
+  password: "",
+  confirmPassword: "",
+};
+const validationSchema = Yup.object({
+  fullName: Yup.string()
+    .max(15, "Must be 15 characters or less")
+    .required("Required"),
+  email: Yup.string().email("Invalid email address").required("Required"),
+  password: Yup.string().required("Required"),
+  confirmPassword: Yup.string().oneOf(
+    [Yup.ref("password"), null],
+    "Passwords must match"
+  ),
+});
+const onSubmit = (values, onSubmitProps) => {
+  alert(JSON.stringify(values, null, 2));
+  onSubmitProps.setSubmitting(false);
+};
```

Comment:
It is best practice to keep ```onSubmit``` function inside component instead of outside component because sometimes we need to use component props inside ```onSubmit``` function.